exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/list/ListMerkleRootCalculator.java

```diff
@@ -44,26 +53,33 @@
 
   private long index;
 
-  private HashCode calculatedRootHash;
+  private HashCode merkleRoot;
 
   /**
-   * Creates a new ListProofRootHashCalculator.
+   * Creates a new ListMerkleRootCalculator.
    *
    * @param serializer a serializer of list elements
    */
-  ListProofRootHashCalculator(ListProofNode listProof, Serializer<E> serializer) {
+  ListMerkleRootCalculator(ListProof listProof, Serializer<E> serializer) {
     this(listProof, serializer, Hashing.defaultHashFunction());
   }
 
-  private ListProofRootHashCalculator(ListProofNode listProof, Serializer<E> serializer,
+  private ListMerkleRootCalculator(ListProof listProof, Serializer<E> serializer,
       HashFunction hashFunction) {
     this.serializer = CheckingSerializerDecorator.from(serializer);
     this.hashFunction = checkNotNull(hashFunction);
     elements = new TreeMap<>();
     index = 0;
-    calculatedRootHash = null;
+    merkleRoot = null;
+
+    ListProofNode rootNode = listProof.getRootNode();
+    rootNode.accept(this);
 
-    listProof.accept(this);
+    merkleRoot = hashFunction.newHasher()
```

Comment:
It turns out to be **not** the Merkle root, but the hash of the collection (what is the term for that?).

Reply:
It's `hash` of the ProofList object in MerkleDB.